### PR TITLE
git issue automation pipeline

### PR DESCRIPTION
**Pull request by omriezra**
This pipeline is created to automate github issues handling:
- Creates jira issue on new creation of github issues, and adds label to the issues with create jira issues (e.g ASM-123)
- Updates jira issue on comments in github issue
- Transitons jira issue to done on closing of github issue

### CODE REVIEW

File: .github/scripts/git_issues/generate_jira_issue_data.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+git_issue_body=$(echo "$GITHUB_CONTEXT" | jq '.event.issue.body' --raw-output)
+git_issue_url=$(echo "$GITHUB_CONTEXT" | jq '.event.issue.html_url' --raw-output)
+git_issue_created_at=$(echo "$GITHUB_CONTEXT" | jq '.event.issue.created_at' --raw-output)
+git_issue_updated_at=$(echo "$GITHUB_CONTEXT" | jq '.event.issue.updated_at' --raw-output)
+git_issue_comment_url=$(echo "$GITHUB_CONTEXT" | jq '.event.comment.html_url' --raw-output)
+git_issue_comment_body=$(echo "$GITHUB_CONTEXT" | jq '.event.comment.body' --raw-output)
+git_issue_comment_created_at=$(echo "$GITHUB_CONTEXT" | jq '.event.comment.created_at' --raw-output)
+git_issue_labels=$(echo "$GITHUB_CONTEXT" | jq '.event.issue.labels[].name' --raw-output)
+
+while IFS='' read -r line; do git_issue_labels_list+=("$line"); done < <(echo "${git_issue_labels}")
```
Comment (OriBenHur-akeyless) [inline .github/scripts/git_issues/generate_jira_issue_data.sh:19]:
```suggestion
git_issue_labels_list=()
while IFS='' read -r line; do git_issue_labels_list+=("$line"); done < <(echo "${git_issue_labels}")
```

File: .github/scripts/git_issues/generate_jira_issue_data.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  echo "Generation new comment on jira issue ${akeyless_jira_issue}"
+  JIRA_EVENT_DESCRIPTION="*Git Issue has new comment*
+  *Git User:* ${git_user}
+  *Git issue comment:*
+  ${git_issue_comment_body}
+  *Git Issue Comment Creation Time:* ${git_issue_comment_created_at}
+  *Git Issue Comment URL:* ${git_issue_comment_url}"
+fi
+
+delimiter="$(openssl rand -hex 8)"
```
Comment (OriBenHur-akeyless) [inline .github/scripts/git_issues/generate_jira_issue_data.sh:62]: Why are you using this delimiter?

**Review by OriBenHur-akeyless — APPROVED** [review]

File: .github/scripts/git_issues/generate_jira_issue_data.sh (hunk 2 above)
Comment (omriezra) [reply .github/scripts/git_issues/generate_jira_issue_data.sh:62]: method that i worked with on slack message the generate random delimiter the context for markdown

Comment (OriBenHur-akeyless) [reply .github/scripts/git_issues/generate_jira_issue_data.sh:62]: But why it's needed?

Comment (omriezra) [reply .github/scripts/git_issues/generate_jira_issue_data.sh:62]: this is preparation for slack message if needed

**Review by niv-ez — APPROVED** [review]

File: .github/scripts/git_issues/generate_jira_issue_data.sh (hunk 1 above)
Comment (omriezra) [reply .github/scripts/git_issues/generate_jira_issue_data.sh:20]: fixed

**Pull request merged**